pkg/discovery/network/pods.go

```diff
@@ -50,18 +50,29 @@ func findPodCommandParameter(clientSet kubernetes.Interface, labelSelector, para
 }
 
 func findPod(clientSet kubernetes.Interface, labelSelector string) (*v1.Pod, error) {
-	pods, err := clientSet.CoreV1().Pods("").List(v1meta.ListOptions{
+	listOptions := v1meta.ListOptions{
 		LabelSelector: labelSelector,
 		Limit:         1,
-	})
+	}
+	pods, err := FindPods(clientSet, listOptions)
+
+	if err != nil || pods == nil {
+		return nil, err
+	}
+
+	return &pods.Items[0], err
+}
+
+func FindPods(clientSet kubernetes.Interface, options v1meta.ListOptions) (*v1.PodList, error) {
+	pods, err := clientSet.CoreV1().Pods("").List(options)
 
 	if err != nil {
-		return nil, errors.WithMessagef(err, "error listing Pods by label selector %q", labelSelector)
+		return nil, errors.WithMessagef(err, "error listing Pods by label selector %q", options.LabelSelector)
 	}
 
 	if len(pods.Items) == 0 {
 		return nil, nil
 	}
 
-	return &pods.Items[0], nil
+	return pods, nil
```

Comment:
It's not exactly the same, pods wouldn't be nil on return then I think.

While on this case the function returns nil,nil

@tpantelis I think that in this case is better not to change the contract of the function since it's used in other places. (network discovery for example)

Reply:
@mangelajo as I understand it, you’re both asking for the same thing — to leave `findPod` as-is. `FindPods` can’t be used anywhere else because it’s new to this PR; there might be other similar functions elsewhere.

Reply:
yes leave `findPod` as-is. I also assert we don't even need `FindPods` - it's basically a thin wrapper for `clientSet.CoreV1().Pods("").List(options)` so I don't think adds any value,

Reply:
FindPods lists ALL pods in the given ns. findPod lists just 1 pod `(Limit: 1)`.
pods in FindPods will be nil when there are no pods matching a label.

The idea to add FindPods is to be able to list pods with varying ListOptions whenever needed.

Reply:
Adjusting findPod to my requirements will need changes in all findPod (removing the `Limit: 1` and returning the whole list and not just pods.Item[0] which in turn will need changes in all the places that calls findPod.

Reply:
My point is that the caller can just call `clientSet.CoreV1().Pods("").List(options)`. `FindPods` adds no value on top of that.  And we wouldn't want to return nil as `List` does not.

Reply:
> Adjusting findPod to my requirements will need changes in all findPod (removing the `Limit: 1` and returning the whole list and not just pods.Item[0] which in turn will need changes in all the places that calls findPod.

right that's why we're saying to leave `findPod` as is.

Reply:
Keeping findpod as is and removing FindPods